### ISSUE TITLE
neotree should be evilified in hybrid mode

### ISSUE THREAD

**Issue by aaronjensen** [syl20bnr/spacemacs#5645]
#### Description
If you are in hybrid mode, neotree is in emacs more rather than evilified

#### Reproduction guide
- Start Emacs
- <kbd>SPC t E h</kbd> to enable hybrid mode
- <kbd>SPC p t</kbd> to open neotree

*Observed behaviour:*
neotree is not evilified

*Expected behaviour:*
neotree is evilified

#### System Info
- OS: darwin
- Emacs: 25.0.92.1
- Spacemacs: 0.105.14
- Spacemacs branch: enable-non-native-fullscreen-without-mac-port (rev. f1220d4)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: helm
- Layers:
```elisp
(better-defaults spacemacs-layouts spacemacs-helm emacs-lisp markdown
                 (syntax-checking :variables syntax-checking-enable-tooltips nil)
                 (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
                 erlang elixir git dash html clojure
                 (org :variables org-enable-github-support t)
                 colors osx github javascript deft floobits ruby
                 (shell :variables shell-default-shell 'ansi-term shell-default-height 30 shell-default-position 'bottom)
                 spell-checking ranger version-control rcirc evil-little-word auto-correct frame-geometry)
```

**Comment by nixmaniack** [comment]
I'm using `hybrid` from start (without toggle `SPC t E h`) which has evilified neotree. So it might be the toggle or evilify macro which isn't handling this dynamic toggle correctly.

**Comment by aaronjensen** [comment]
I too use hybrid from the start and it doesn't work. I'm on develop.

**Comment by bmag** [comment]
I have this in my config, and neotree is evilified:
```elisp
   dotspacemacs-editing-style '(hybrid :variables
                                       hybrid-mode-default-state 'normal
                                       hybrid-mode-enable-evilified-state t
                                       hybrid-mode-enable-hjkl-bindings nil)
```
But I'm running develop from about a week ago (rev. a9c9e9a).

**Comment by nixmaniack** [comment]
What is it that's missing, a bit more specific? What is the value of `M-:` `evil-state` `RET` when you're in neotree buffer?

**Comment by nixmaniack** [comment]
Oh right, I also have what @bmag has specified. These variables allow to move your hybrid style more towards emacs or vim depending on your liking.

**Comment by aaronjensen** [comment]
Ah, thanks. That works for me. I left a comment on #5484 as this should maybe be the default. @syl20bnr feel free to close this if you disagree or want to track it in the issue.

**Comment by syl20bnr** [comment]
We can set the default to t for evilified state.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
